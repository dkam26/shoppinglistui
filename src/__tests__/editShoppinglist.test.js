import React from 'react';
import { shallow, mount } from 'enzyme';
import {configure} from 'enzyme';
import Editshoppinglist from '../components/edit.shoppinglist.component';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import toJson,{ shallowToJson } from 'enzyme-to-json';
import LocalStorageMock from '../setupTests';
import ReactDOM from 'react-dom';
configure({ adapter: new Adapter() });
describe('edit shoopinglists component', () => {
     const wrapper =  shallow(<Editshoppinglist />);
    it('should run', () => {
        expect(wrapper.exists(<div></div>)).toBe(true);
    });
    it('should render correctly', () => {
        const output = shallow( <Editshoppinglist />);
        expect(shallowToJson(output)).toMatchSnapshot();
        });
        it('should edit shoppinglist',()=>{
            wrapper.find('#EditShoppinglist').simulate('click')
            expect(toJson(wrapper)).toMatchSnapshot();
           
            
        });
    it('renders without crashing', () => {
            const div = document.createElement('div');
            ReactDOM.render(<Editshoppinglist />, div);
        });   
      
})