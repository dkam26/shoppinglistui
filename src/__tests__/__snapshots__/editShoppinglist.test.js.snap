// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit shoopinglists component should edit shoppinglist 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        >
          <MenuItem
            name="logout"
            style={
              Object {
                "color": "white",
              }
            }
          />
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuMenu
            position="right"
          >
            <MenuItem>
              <Button
                as="button"
                content="Back"
                icon="left chevron"
                labelPosition="left"
                onClick={[Function]}
              />
            </MenuItem>
          </MenuMenu>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Edit  
               list
            </h3>
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <div
      className="login-form"
    >
      <style>
        
      body &gt; div,
      body &gt; div &gt; div,
      body &gt; div &gt; div &gt; div.login-form {
        height: 100%;
      }
    
      </style>
      <Grid
        style={
          Object {
            "height": "100%",
          }
        }
        textAlign="center"
        verticalAlign="middle"
      >
        <GridColumn
          style={
            Object {
              "maxWidth": 450,
            }
          }
        >
          <Form
            as="form"
            size="large"
          >
            <Segment
              stacked={true}
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                name="shoppinglist"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
              <Button
                as="button"
                color="blue"
                fluid={true}
                id="EditShoppinglist"
                onClick={[Function]}
                size="large"
              >
                Rename shoppinglist
              </Button>
            </Segment>
          </Form>
        </GridColumn>
      </Grid>
    </div>
  </Container>
  <_class
    options={Object {}}
  />
</div>
`;

exports[`edit shoopinglists component should render correctly 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        >
          <MenuItem
            name="logout"
            style={
              Object {
                "color": "white",
              }
            }
          />
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuMenu
            position="right"
          >
            <MenuItem>
              <Button
                as="button"
                content="Back"
                icon="left chevron"
                labelPosition="left"
                onClick={[Function]}
              />
            </MenuItem>
          </MenuMenu>
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Edit  
               list
            </h3>
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <div
      className="login-form"
    >
      <style>
        
      body &gt; div,
      body &gt; div &gt; div,
      body &gt; div &gt; div &gt; div.login-form {
        height: 100%;
      }
    
      </style>
      <Grid
        style={
          Object {
            "height": "100%",
          }
        }
        textAlign="center"
        verticalAlign="middle"
      >
        <GridColumn
          style={
            Object {
              "maxWidth": 450,
            }
          }
        >
          <Form
            as="form"
            size="large"
          >
            <Segment
              stacked={true}
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                name="shoppinglist"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
              <Button
                as="button"
                color="blue"
                fluid={true}
                id="EditShoppinglist"
                onClick={[Function]}
                size="large"
              >
                Rename shoppinglist
              </Button>
            </Segment>
          </Form>
        </GridColumn>
      </Grid>
    </div>
  </Container>
  <_class
    options={Object {}}
  />
</div>
`;
